doc/user-guide/terminology.rst

```diff
@@ -112,3 +112,63 @@ complete examples, please consult the relevant documentation.*
         ``__array_ufunc__`` and ``__array_function__`` protocols are also required.
 
         __ https://numpy.org/neps/nep-0022-ndarray-duck-typing-overview.html
+
+    Aligning
+
+        Aligning refers to the process of making sure that the dimensions
+        and coordinates of two or more DataArrays or Datasets are consistent with each
+        other so that they can be combined or compared properly.
+
```

Comment:
I think this should mention that missing values after alignment are filled with np.nan.